src/riscv.c

```diff
@@ -143,10 +155,15 @@ bool rv_enables_to_output_exit_code(riscv_t *rv)
     return rv->output_exit_code;
 }
 
+
 void rv_delete(riscv_t *rv)
 {
     assert(rv);
+#if !RV32_HAS(JIT)
     block_map_destroy(rv);
+#else
+    cache_free(rv->block_cache, NULL);
```

Comment:
The second parameter is set to `NULL`, but there is no check for callback in the implementation of `cache_free`, resulting in segmentation fault.
Send another pull request to fix prior to the merge of this pull request.